adr/ADR-8.md

```diff
@@ -214,6 +215,44 @@ type KV interface {
 }
 ```
 
+## KV Management
+
+This is set of operations on the KV stores from the JetStream context.
+
+```go
+// KeyValueManager is used to manage KeyValue stores. It provides methods to
+// create, delete, and retrieve.
+type KeyValueManager interface {
+    // KeyValue will lookup and bind to an existing KeyValue store.
+    // Name can be `get_key_value`, or whatever name is idiomatic in given language.
+    KeyValue(ctx context.Context, bucket string) (KeyValue, error)
+
+    // CreateKeyValue will create a KeyValue store with the given
+    // configuration.
+    CreateKeyValue(ctx context.Context, cfg KeyValueConfig) (KeyValue, error)
+
+    // UpdateKeyValue will update an existing KeyValue store with the given
+    // configuration.
+    UpdateKeyValue(ctx context.Context, cfg KeyValueConfig) (KeyValue, error)
+
+    // CreateOrUpdateKeyValue will create a KeyValue store if it does not
+    // exist or update an existing KeyValue store with the given
+    // configuration (if possible).
+    CreateOrUpdateKeyValue(ctx context.Context, cfg KeyValueConfig) (KeyValue, error)
```

Comment:
If you have create/update behave correctly, there's no need for create update - that is just lazy api to simplify the checking without actually knowing what it did - this further propagates the pattern that shouldn't be there.